Code/Mantid/Framework/RemoteJobManagers/src/LSFJobManager.cpp

```diff
@@ -0,0 +1,1257 @@
+#include <fstream>
+
+#include "MantidKernel/Exception.h"
+#include "MantidKernel/InternetHelper.h"
+#include "MantidKernel/Logger.h"
+#include "MantidRemoteJobManagers/LSFJobManager.h"
+
+#include <boost/lexical_cast.hpp>
+#include "boost/algorithm/string/replace.hpp"
+
+#include <Poco/AutoPtr.h>
+#include <Poco/File.h>
+#include <Poco/DOM/Document.h>
+#include <Poco/DOM/DOMParser.h>
+#include <Poco/DOM/Element.h>
+#include <Poco/DOM/NodeList.h>
+#include <Poco/Net/HTTPRequest.h>
+#include <Poco/StreamCopier.h>
+
+namespace Mantid {
+namespace RemoteJobManagers {
+
+// Do not declare as remote job manager for the factory, this is an abstract
+// class
+// DECLARE_REMOTEJOBMANAGER()
+
+namespace {
+// static logger object
+Mantid::Kernel::Logger g_log("LSFJobManager");
+}
+
+std::string LSFJobManager::m_killPathBase =
+    "webservice/pacclient/jobOperation/kill/";
+
+std::string LSFJobManager::m_allJobsStatusPath = "webservice/pacclient/jobs?";
+
+std::string LSFJobManager::m_jobIdStatusPath = "webservice/pacclient/jobs/";
+
+//  The 0 at the end of the upload path is 'jobId' 0, if a jobId is given the
+//  upload goes to a path relative to the job path.
+std::string LSFJobManager::m_uploadPath = "webservice/pacclient/upfile/0";
+
+std::string LSFJobManager::m_submitPath = "webservice/pacclient/submitapp";
+
+std::string LSFJobManager::m_downloadOneBasePath = "webservice/pacclient/file/";
+
+std::string LSFJobManager::m_downloadAllJobFilesBasePath =
+    "webservice/pacclient/jobfiles/";
+
+std::string LSFJobManager::m_acceptType = "text/plain,application/xml,text/xml";
+
+using namespace Mantid::Kernel;
+
+void LSFJobManager::abortRemoteJob(const std::string &jobID) {
+  if (m_tokenStash.size() <= 0) {
+    throw std::runtime_error(
+        "Job status query failed. You do not seem to have logged "
+        "in.");
+  }
+  // only support for single-user
+  Token tok = m_tokenStash.begin()->second;
+  const std::string killPath = m_killPathBase + jobID;
+  const std::string baseURL = tok.m_url;
+  const std::string token = tok.m_token_str;
+
+  std::string httpsURL = baseURL + killPath;
+  StringToStringMap headers;
```

Comment:
I'll check how this could be done. I would be nice to have a better representation of the URIs. But at first sight it seems that these lines can get confusing with Poco::URI. The path component is split between a base url that you get from SCARF when you authenticate and a second part. And the paths are subject to change at the moment.